### PR TITLE
Replace tslint with typescript-eslint

### PR DESCRIPTION
**Pull request by diksipav**

### CODE REVIEW

**Review by scotttrinh — COMMENTED** [review]
Looks good!

File: package.json [hunk 1]
```diff
@@ -7,19 +7,15 @@
...
-    "tslint-config-prettier": "^1.18.0",
-    "tslint-plugin-prettier": "^2.3.0",
-    "typescript": "^5.4.5"
+    "typescript": "^5.4.5",
+    "typescript-eslint": "^7.8.0"
   },
   "scripts": {
-    "lint": "tslint 'packages/*/src/**/*.ts'",
-    "eslint": "eslint 'packages/*/src/**/*.ts'",
+    "lint": "eslint 'packages/*/src/**/*.ts'",
```
[the remark this answers is not in the document]
Comment (diksipav) [reply package.json:17]: oh yes yes sorry!

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      "@typescript-eslint/no-explicit-any": "off",
+      "@typescript-eslint/no-empty-function": "warn",
+      "@typescript-eslint/ban-ts-comment": "warn",
+      "@typescript-eslint/no-unused-vars": [
+        "error",
+        { argsIgnorePattern: "^_", varsIgnorePattern: "^_" },
+      ],
+      // Stylistic rules
+      // We need to use the tseslint.configs.stylistic, but some stylistic rules are erroring.
+      // These 4 pass. We can gradually add the rest and at the end replace with the tseslint config.
```
[the remark this answers is not in the document]
Comment (diksipav) [reply eslint.config.mjs:31]: oh, I am not completely sure what exactly is breaking (except for one rule), alos there are hundreds of changes and only some of them is making a problem... but yes, I can do that, just wanted to finish asap : )

[the remark this answers is not in the document]
Comment (diksipav) [reply eslint.config.mjs:31]: and I mean this can be tracked down, just I need maybe 2 more h...

[the remark this answers is not in the document]
Comment (scotttrinh) [reply eslint.config.mjs]: Don't sweat it for now, these rules are already beyond what we had in tslint, so I'm not super worried.

**Review by scotttrinh — APPROVED** [review]
👏 Looks good! Can you update the title of this PR before merging to something like: "Replace tslint with typescript-eslint".

**Pull request merged**